### ISSUE TITLE
Service not found for type 'PlanetXamarin.Portable.Interfaces.IMessage'

### ISSUE THREAD

**Issue by K232** [jamesmontemagno/PlanetXamarin#1]
Hi James,
I'm playing around with your code using the WinStore-Part. I've seen a (for me) strange error: In the _MasterViewModel.cs_ within _public async Task ExecuteLoadItemsCommand()_ you have send out a message:

```
var message = ServiceContainer.Resolve<IMessage>(); 
message.SendMessage("SUCCESS!! HOUSTON IS AWESOME!"); 
```

That's working fine. But when I want to move the message to the top of this procedure I get the following error:

_An exception of type 'System.Collections.Generic.KeyNotFoundException' occurred in mscorlib.dll but was not handled in user code_

_Additional information: Service not found for type 'PlanetXamarin.Portable.Interfaces.IMessage''_

It seems the breaking point is `var responseString = await httpClient.GetStringAsync(feed);`. When I place the message **before** this line, I get the above error. When I place the message *_after_ this line, everything works fine....? Any ideas about how I could display a message already at the beginning?
Mayn thanks!

**Comment by jamesmontemagno** [comment]
I don't do this anywhere in my code. I don't use the servicecontainer at all. See this search: https://github.com/jamesmontemagno/PlanetXamarin/search?utf8=%E2%9C%93&q=ServiceContainer

Everything was changed to a different implementation a long time ago.

**Comment by K232** [comment]
Hi James,
you're right. But I just downloaded the code 2 weeks ago so I wonder where it came from.... 

I've found it: I took the code from your link at https://forums.xamarin.com/discussion/11576/portable-class-library-deep-dive which leads to https://github.com/jamesmontemagno/PlanetXamarin/tree/HoustonPCLDeepDive which includes https://github.com/jamesmontemagno/PlanetXamarin/blob/HoustonPCLDeepDive/PlanetXamarin.Portable/ViewModels/MasterViewModel.cs and there it is at line 78/79. 

So I hope there were not too many changes as I used this code as a starting point now. 

Anyway, if someone has the same problem, here's an easy fix: Just save the message to be shown in a separate variable and at the end of the code check whether it's an empty string.
